tests/IntegrityTests/ValidateReferencedPartials.cs

```diff
@@ -0,0 +1,67 @@
+﻿using Microsoft.VisualBasic;
+using NUnit.Framework;
+using NUnit.Framework.Internal;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace IntegrityTests
+{
+    internal class ValidateReferencedPartials
+    {
+        [Test]
+        public void ReferencedPartialsAreUsed()
+        {           
+            var allErrors = new Dictionary<string, List<string>>();
+            var allPartials = Directory.GetFiles(TestSetup.DocsRootPath, "*.partial.md", SearchOption.AllDirectories);
+            var allArticles = Directory.GetFiles(TestSetup.DocsRootPath, "*.md", SearchOption.AllDirectories).Except(allPartials);
+
+            foreach (var articlePath in allArticles)
+            {
+                //Skips the readme file because it references a partial as instructions
+                if (articlePath.Contains("README.md"))
+                {
+                    continue;
+                }
+
+                var fileName = Path.GetFileNameWithoutExtension(articlePath);
+                string text = File.ReadAllText(articlePath);
+                string searchPattern = @"(?i)partial:\s*[a-zA-Z0-9-_]*";
+                string replacePattern = @"(?i)partial:\s*";
+                var matches = Regex.Matches(text, searchPattern, RegexOptions.IgnoreCase);
```

Comment:
This makes sure `partial` must occur at the beginning of a line, to guard against the (probably rare) case that `partial: ` would occur in the middle of valid text.

Also `_` can't be in a partial name since that's the filename delimiter.
```suggestion
                string searchPattern = @"^(?i)partial:\s*[a-zA-Z0-9-]*";
                string replacePattern = @"(?i)partial:\s*";
                var matches = Regex.Matches(text, searchPattern, RegexOptions.IgnoreCase | RegexOptions.Multiline);
```